src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -133,13 +134,7 @@ internal class ConnectionManager : IConnectionManager {
             var isRemoved = _connections.TryRemove(name, out connection);
             if (isRemoved) {
                 UpdateRecentConnections();
-
-                // Credentials are saved by URI. Delete the credentials if there are no other connections using it.
-                if (_connections.All(kvp => kvp.Value.Uri != connection.Uri)) {
-                    if (connection.Uri != null) {
-                        _securityService.DeleteUserCredentials(connection.Uri.ToCredentialAuthority());
-                    }
-                }
+                _securityService.DeleteUserCredentials(connection.ToCredentialAuthority());
```

Comment:
`connection.BrokerConnectionInfo.CredentialAuthority`

And you can remove `ToCredentialAuthority` with its class.